components/backup-stream/src/endpoint.rs

```diff
@@ -869,7 +878,8 @@ where
         })
     }
 
-    fn on_flush_with_min_ts(&self, task: String, min_ts: TimeStamp) {
+    fn on_flush_with_min_ts(&mut self, task: String, min_ts: TimeStamp) {
+        self.checkpoint_mgr.freeze();
```

Comment:
what if `on_flush_with_min_ts` happen in a very short time. will the second freeze clear the state of the first one?

Reply:
The `flush` for each task will be guarded by an atomic value. Which makes sure there won't be concurrent flushes.